### PR TITLE
PCX-2334: Programmatically create versionName for snapshot versions

### PR DESCRIPTION
**Pull request by siebeprojects**
**Description**
AS a developer
I LIKE to automatically detect if the Android SDK is a snapshot version
SO THAT I do not manually have to modify the VERSIONNAME file

**Acceptance Criteria**
- The version of the Checkout SDK is only a release version for master branch
- For all other branches the Checkout SDK has a snapshot version
- The versionName ends with "-SNAPSHOT" for snapshot versions

### CODE REVIEW

File: checkout/publish.gradle [hunk 1]
```diff
@@ -3,7 +3,7 @@ task publishSnapshot {
         logger.lifecycle('[PUBLISH-SNAPSHOT] Missing nexusUser and/or nexusPassword environment variables')
         return
     }
-    if (rootProject.isSnapshotVersion() == false) {
+    if (rootProject.isReleaseVersion() == true) {
```
Comment (Fbada006) [inline checkout/publish.gradle:6]: I believe there is no need to check it like this. 
```java
if (rootProject.isReleaseVersion()) {}
```
works just fine

File: checkout/publish.gradle [hunk 2]
```diff
@@ -30,15 +30,15 @@ afterEvaluate {
         onlyIf {
             ((repository == publishing.repositories.nexusSnapshots) &&
                     (publication == publishing.publications.nexusSnapshots) &&
-                    (rootProject.isSnapshotVersion() == true)) ||
+                    (rootProject.isReleaseVersion() == false)) ||
 
                     ((repository == publishing.repositories.nexusReleases) &&
                             (publication == publishing.publications.nexusReleases) &&
-                            (rootProject.isSnapshotVersion() == false)) ||
+                            (rootProject.isReleaseVersion() == true)) ||
```
Comment (Fbada006) [inline checkout/publish.gradle:37]: Same here

File: checkout/publish.gradle [hunk 3]
```diff
@@ -30,15 +30,15 @@ afterEvaluate {
...
 
                     ((repository == publishing.repositories.packagecloud) &&
                             (publication == publishing.publications.packagecloud) &&
-                            (rootProject.isSnapshotVersion() == false))
+                            (rootProject.isReleaseVersion() == true))
```
Comment (Fbada006) [inline checkout/publish.gradle:41]: And here.

File: checkout/publish.gradle [hunk 4]
```diff
@@ -19,7 +19,7 @@ task publishRelease {
         logger.lifecycle('[PUBLISH-RELEASE] Missing packageCloudToken environment variable')
         return
     }
-    if (rootProject.isSnapshotVersion() == true) {
+    if (rootProject.isReleaseVersion() == false) {
```
Comment (Fbada006) [inline checkout/publish.gradle:22]: To follow the above pattern, we can use 
```java
    if (!rootProject.isReleaseVersion()) {
```

File: checkout/publish.gradle (hunk 1 above)
Comment (siebeprojects) [reply checkout/publish.gradle]: Removed the duplicated boolean check

File: checkout/publish.gradle (hunk 2 above)
Comment (siebeprojects) [reply checkout/publish.gradle]: Removed the duplicated boolean check

File: checkout/publish.gradle (hunk 3 above)
Comment (siebeprojects) [reply checkout/publish.gradle]: Removed the duplicated boolean check

File: checkout/publish.gradle (hunk 4 above)
Comment (siebeprojects) [reply checkout/publish.gradle]: Removed the duplicated boolean check

**Review by Fbada006 — APPROVED** [review]

**Pull request merged**